spec.bs

```diff
@@ -1387,6 +1478,227 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
+      * |maybeKey| is failure, or its [=byte sequence/length=] is not 32;
+      * |maybeSignature| is failure, or its [=byte sequence/length=] is not 64;
+    1. Let |signature| be a [=signed additional bid signatures=], whose
+      [=signed additional bid signature/key=] is |maybeKey|, and
+      [=signed additional bid signature/signature=] is |maybeSignature|.
+    1. [=list/Append=] |signature| to |signatures|.
+  1. If |decodeSignatureFailed| is true, then return null.
+  1. Let |decodedAdditionalBid| be the result of [=decode an additional bid=] given
+    |parsedSignedAdditionalBid|["bid"], |auctionConfig| and |topLevelAuctionConfig|.
+  1. Return null if any of the following conditions hold:
+    * |decodedAdditionalBid| is failure;
+    * The result of [=checking a currency tag=] with |decodedAdditionalBid|'s
+      [=additional bid/bid=]'s [=generated bid/bid=]'s [=bid with currency/currency=] and the
+      result of running [=look up per-buyer currency=] with |auctionConfig|.
+  1. Check whether the additional bid is negative targeted:
+    1. Let |verifiedSignatureKeys| be a new [=set=] of [=byte sequences=].
+    1. [=list/For each=] |signature| of |signatures|:
+      1. Let |isSignatureValid| be the result of running [=verify=] |signature|'s
+        [=signed additional bid signature/signature=] on message |parsedSignedAdditionalBid|["bid"]
+        using |signature|'s [=signed additional bid signature/key=], with 0 for Ed25519ctx.
+      1. If |isSignatureValid| is true:
+        1. [=set/Append=] |signature|'s [=signed additional bid signature/key=] to
+          |verifiedSignatureKeys|.
+    1. Let |negativeTargeted| be false.
+    1. Let |additionalBidBuyer| be |additionalBid|'s [=additional bid/bid=]'s
+      [=generated bid/interest group=]'s [=interest group/owner=].
+    1. [=list/For each=] |igName| of |additionalBid|'s
+      [=additional bid/negative target interest group names=]:
+      1. If |negativeTargetInfo|[(|additionalBidBuyer|, |igName|)] [=map/exists=]:
+        1. Let (|joiningOrigin|, |additionalBidKey|) be
+          |negativeTargetInfo|[(|additionalBidBuyer|, |igName|)].
+        1. If |verifiedSignatureKeys| [=list/contains=] |additionalBidKey|:
+          1. If |joiningOrigin| is not null:
+            1. If |joiningOrigin| is not [=same origin=] with |additionalBid|'s
+              [=additional bid/negative target joining origin=], then [=iteration/continue=].
+          1. Set |negativeTargeted| to true, and [=iteration/break=].
+
+     Note: If the signature doesn't verify successfully, the additional bid proceeds as if the
+        negative interest group is not present. This ensures that only the owner of the negative
+        interest group, who created the additonalBidKey, is allowed to negatively target the
+        interest group, and that nobody else can learn whether the interest group is present on
+        the device.
+  1. If |negativeTargeted| is true:
+    1. Return null.

```

Comment:
Similar to some feedback above, not every if body needs to be in its own nested list of steps. This would be much more readable on a single line.